File: app/src/main/java/com/aware/app/testx/presenter/SymptomAdapter.java
package com.aware.app.testx.presenter;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import com.aware.app.testx.R;

import java.util.Arrays;

public class SymptomAdapter extends BaseAdapter {

    private Context context;
    private TypedArray symptoms;
    private int[] result;

    public SymptomAdapter(Context context) {
        this.context = context;
        symptoms = context.getResources().obtainTypedArray(R.array.symptomsList);

        result = new int[symptoms.length()];
        for (int i=0; i<result.length; i++) result[i] = -1;
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {

        View listItem = convertView;
        if (listItem == null) {
            listItem = LayoutInflater.from(context).inflate(R.layout.view_list_item_symptom, parent, false);
        }

        TextView symptomText = listItem.findViewById(R.id.symptomName);
        RadioGroup group = listItem.findViewById(R.id.symptomRate);
        RadioButton rate0 = listItem.findViewById(R.id.rate0);
        RadioButton rate1 = listItem.findViewById(R.id.rate1);
        RadioButton rate2 = listItem.findViewById(R.id.rate2);
        RadioButton rate3 = listItem.findViewById(R.id.rate3);
        RadioButton rate4 = listItem.findViewById(R.id.rate4);

        String[] symptom = context.getResources().getStringArray(symptoms.getResourceId(position, -1));
        symptomText.setText(symptom[0]);
        rate0.setText(symptom[1]);
        rate1.setText(symptom[2]);
        rate2.setText(symptom[3]);
        rate3.setText(symptom[4]);
        rate4.setText(symptom[5]);

        group.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                result[position] = group.indexOfChild(group.findViewById(checkedId));
            }
        });

        return listItem;
    }

    @Override
    public int getCount() {
        return symptoms.length();
    }

    @Override
    public String[] getItem(int position) {
        return context.getResources().getStringArray(symptoms.getResourceId(position, -1));
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public int getViewTypeCount() {
        return symptoms.length();
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    public int getChecked(int position) {
        return result[position];
    }
}

File: app/src/main/java/com/aware/app/testx/presenter/SymptomAdapter.kt
package com.aware.app.testx.presenter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.aware.app.testx.R
import kotlinx.android.synthetic.main.view_list_item_symptom.view.*
import java.util.*

class SymptomAdapter(private val context: Context) : RecyclerView.Adapter<SymptomAdapter.SymptomHolder>() {

    // get array of symptoms description from resource file
    private val symptoms = context.resources.obtainTypedArray(R.array.symptomsList)

    // array for storing self-evaluation values
    var result = IntArray(symptoms.length())

    inner class SymptomHolder(view: View) : RecyclerView.ViewHolder(view) {
        val symptomText: TextView = view.symptomName
        val group: RadioGroup = view.symptomRate
        var rate0: RadioButton = view.findViewById(R.id.rate0)
        var rate1: RadioButton = view.findViewById(R.id.rate1)
        var rate2: RadioButton = view.findViewById(R.id.rate2)
        var rate3: RadioButton = view.findViewById(R.id.rate3)
        var rate4: RadioButton = view.findViewById(R.id.rate4)
    }

    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
        super.onAttachedToRecyclerView(recyclerView)

        // set default assessment values as -1
        Arrays.fill(result, -1)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SymptomHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.view_list_item_symptom, parent, false)

        return SymptomHolder(view)
    }

    override fun onBindViewHolder(holder: SymptomHolder, position: Int) {
        // verify that radiogroup has correct checked parameter
        holder.group.setOnCheckedChangeListener(null)
        holder.group.clearCheck()
        if (result[position] != -1) holder.group.check(holder.group.getChildAt(result[position]).id)

        // label views from resource file
        val symptom = context.resources.getStringArray(symptoms.getResourceId(position, -1))
        holder.symptomText.text = symptom[0]
        holder.rate0.text = symptom[1]
        holder.rate1.text = symptom[2]
        holder.rate2.text = symptom[3]
        holder.rate3.text = symptom[4]
        holder.rate4.text = symptom[5]

        // record assessment selection
        holder.group.setOnCheckedChangeListener { group, checkedId ->
                result[position] = group.indexOfChild(group.findViewById(checkedId))
        }
    }

    override fun getItemCount(): Int {
        return symptoms.length()
    }

    fun getName(position: Int): String {
        return context.resources.getStringArray(symptoms.getResourceId(position, -1))[0]
                .replace(" ".toRegex(), "_").toLowerCase(Locale.getDefault())
    }
}
